src/main/webapp/js/components/fields/field-config-modal.jsx

```diff
@@ -0,0 +1,260 @@
+  };
+
+  displayFields = (item) => {
+    const {
+      base, editable, filterable, required, fieldType, entity,
+      displayName, name, format, relatedEntity, relatedField, openConfirmModal,
+    } = this.state;
+    return (
+      !!item && (
+      <div className="config-modal-form">
+        <Form
+          onSubmit={this.onSubmit}
+          initialValues={{
+            base,
+            fieldType,
+            entity,
+            relatedEntity,
+            editable,
+            filterable,
+            required,
+            displayName,
+            name,
+            format,
+            relatedField,
+          }}
+          render={({ handleSubmit }) => (
+            <form onSubmit={handleSubmit} className="modal-fields">
+              { renderFormField({
+                name: 'base', fieldType: 'BOOLEAN', displayName: 'Base', required: true,
+              }) }
+              { renderFormField({
+                name: 'editable', fieldType: 'BOOLEAN', displayName: 'Editable', required: true,
+              }) }
+              { renderFormField({
+                name: 'filterable', fieldType: 'BOOLEAN', displayName: 'Filterable', required: true,
+              }) }
+              { renderFormField({
+                name: 'required', fieldType: 'BOOLEAN', displayName: 'Required', required: true,
+              }) }
+              { renderFormField({
+                name: 'fieldType', fieldType: 'RELATION', displayName: 'Field type', required: true, options: FIELD_TYPE_OPTIONS,
+              }) }
+              { renderFormField({
+                name: 'entity', fieldType: 'RELATION', displayName: 'Entity', required: true, options: ENTITY_TYPE_OPTION,
+              }) }
+              { renderFormField({
+                name: 'relatedEntity', fieldType: 'RELATION', displayName: 'Related entity', required: true, options: ENTITY_TYPE_OPTION,
+              }) }
+              { renderFormField({
+                name: 'displayName', fieldType: 'TEXT', displayName: 'Display name', required: true,
+              }) }
+              { renderFormField({
+                name: 'name', fieldType: 'TEXT', displayName: 'Name', required: true,
+              }) }
+              { renderFormField({
+                name: 'format', fieldType: 'TEXT', displayName: 'Format', required: true,
+              }) }
+              { renderFormField({
+                name: 'relatedField', fieldType: 'TEXT', displayName: 'Related field', required: true,
+              }) }
+              <div>
+                <span>
+                  <button type="submit" className="btn btn-primary">Save</button>
+                </span>
+                <span style={{ float: 'right' }}>
+                  <button type="button" className="btn btn-danger" onClick={(event) => { event.preventDefault(); this.setState(prev => ({ openConfirmModal: !prev.openConfirmModal })); }}>Delete</button>
```

Comment:
you can just set the openConfirmModal to true, I think you should add a separate function to call here and open the modal